### PR TITLE
[15.0][MIG] hr_grade_rank: migrated module

### PR DESCRIPTION
**Pull request by ilakkia-serpentcs**

### CODE REVIEW

**Comment by JayVora-SerpentCS** [comment]
@fetcheping Pls review

File: hr_grade_rank/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# See LICENSE file for full copyright and licensing details.
+
+{
+    "name": "HR-Grade Rank",
+    "version": "14.0.1.0.0",
```
Comment (fetcheping) [inline hr_grade_rank/__manifest__.py]:
```suggestion
    "version": "15.0.1.0.0",
```

File: hr_grade_rank/models/hr_grade_rank.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+from odoo import api, fields, models
+
+
+class RankRank(models.Model):
+    _name = "rank.rank"
+    _description = "Rank"
+
+    name = fields.Char("Name")
+    description = fields.Text("Description")
+    salary_range = fields.Text("Salary Range")
```
Comment (fetcheping) [inline hr_grade_rank/models/hr_grade_rank.py]: @ilakkia-serpentcs, you will have this warning (The attribute string is redundant. String parameter equal to name of variable) when you will run precommit.
so you can do like that:
name = fields.Char()
description = fields.Text(),
salary_range = fields.Text()
so no need to put again string value in this case.

File: hr_grade_rank/views/hr_grade_rank_view.xml [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            </form>
+        </field>
+    </record>
+
+    <!-- GRADE TREE VIEW -->
+    <record id="view_employee_grade_tree" model="ir.ui.view">
+        <field name="name">grade.grade.tree</field>
+        <field name="model">grade.grade</field>
+        <field name="arch" type="xml">
+            <tree string="Grades">
```
Comment (fetcheping) [inline hr_grade_rank/views/hr_grade_rank_view.xml]:
```suggestion
            <tree>
```

File: hr_grade_rank/views/hr_grade_rank_view.xml [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+            </form>
+        </field>
+    </record>
+
+    <!-- RANK TREE VIEW -->
+    <record id="view_employee_rank_tree" model="ir.ui.view">
+        <field name="name">rank.rank.tree</field>
+        <field name="model">rank.rank</field>
+        <field name="arch" type="xml">
+            <tree string="Ranks">
```
Comment (fetcheping) [inline hr_grade_rank/views/hr_grade_rank_view.xml]:
```suggestion
            <tree>
```

File: hr_grade_rank/views/hr_grade_rank_view.xml [hunk 5]
```diff
@@ -0,0 +1,159 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (fetcheping) [inline hr_grade_rank/views/hr_grade_rank_view.xml]: @ilakkia-serpentcs,  you can remove this line

File: hr_grade_rank/__manifest__.py (hunk 1 above)
Comment (ilakkia-serpentcs) [reply hr_grade_rank/__manifest__.py]: updated

File: hr_grade_rank/views/hr_grade_rank_view.xml (hunk 3 above)
Comment (ilakkia-serpentcs) [reply hr_grade_rank/views/hr_grade_rank_view.xml]: removed string

File: hr_grade_rank/views/hr_grade_rank_view.xml (hunk 5 above)
Comment (ilakkia-serpentcs) [reply hr_grade_rank/views/hr_grade_rank_view.xml]: removed as mentioned

File: hr_grade_rank/views/hr_grade_rank_view.xml (hunk 4 above)
Comment (ilakkia-serpentcs) [reply hr_grade_rank/views/hr_grade_rank_view.xml]: removed string

File: hr_grade_rank/models/hr_grade_rank.py (hunk 2 above)
Comment (ilakkia-serpentcs) [reply hr_grade_rank/models/hr_grade_rank.py]: yes  removed string as it is deprecated

**Comment by Trivedi-Vacha-SerpentCS** [comment]
Ready for QA

**Comment by ketanserpentcs** [comment]
This PR testing is assigned to Siddharaj.

**Comment by JayVora-SerpentCS** [comment]
Siddhraj pase thi koi update?

**Comment by ketanserpentcs** [comment]
Now, He is working on it for issue retesting phase. will complete asap

File: hr_grade_rank/models/hr_grade_rank.py [hunk 6]
```diff
@@ -10,7 +10,11 @@ class RankRank(models.Model):
     name = fields.Char()
     description = fields.Text()
     salary_range = fields.Text()
-    grade_id = fields.Many2one("grade.grade", "Grade")
+    grade_id = fields.Many2one("grade.grade",  "Grade")
+    active = fields.Boolean('Active',default=True)
+    _sql_constraints=[('name_uniq', 'unique(name)',"Rank should be unique")]
+
+
```
Comment (Trivedi-Vacha-SerpentCS) [inline hr_grade_rank/models/hr_grade_rank.py]: remove blank spaces

File: hr_grade_rank/models/hr_grade_rank.py [hunk 7]
```diff
@@ -10,7 +10,11 @@ class RankRank(models.Model):
     name = fields.Char()
     description = fields.Text()
     salary_range = fields.Text()
-    grade_id = fields.Many2one("grade.grade", "Grade")
+    grade_id = fields.Many2one("grade.grade",  "Grade")
```
Comment (Trivedi-Vacha-SerpentCS) [inline hr_grade_rank/models/hr_grade_rank.py]: after comma it should have one space only

File: hr_grade_rank/models/hr_grade_rank.py [hunk 8]
```diff
@@ -20,6 +24,9 @@ class GradeGrade(models.Model):
     name = fields.Char()
     description = fields.Text()
     rank_ids = fields.One2many("rank.rank", "grade_id", "Ranks")
+    active = fields.Boolean('Active',default=True)
+
+
```
Comment (Trivedi-Vacha-SerpentCS) [inline hr_grade_rank/models/hr_grade_rank.py]: remove blank lines

File: hr_grade_rank/models/hr_grade_rank.py [hunk 9]
```diff
@@ -37,3 +45,9 @@ def _onchange_grade(self):
             res["domain"] = {"rank_id": [
                 ("id", "in", self.grade_id.rank_ids.ids)]}
         return res
+
+
+    
+
+            
+
```
Comment (Trivedi-Vacha-SerpentCS) [inline hr_grade_rank/models/hr_grade_rank.py]: remove blank lines

File: hr_grade_rank/views/hr_grade_rank_view.xml [hunk 10]
```diff
@@ -88,6 +88,9 @@
         <field name="res_model">rank.rank</field>
         <field name="view_mode">tree,form</field>
     </record>
+   
+
+
```
Comment (Trivedi-Vacha-SerpentCS) [inline hr_grade_rank/views/hr_grade_rank_view.xml:93]: remove blank lines

**Review by Trivedi-Vacha-SerpentCS — CHANGES_REQUESTED** [review]

**Comment by Savitri-SerpentCS** [comment]
> Siddhraj pase thi koi update?

This PR testing has been completed by Siddharaj Vaghela.

**Comment by JayVora-SerpentCS** [comment]
Good to go ?

**Comment by Trivedi-Vacha-SerpentCS** [comment]
@JayVora-SerpentCS ready to merge

**Pull request merged**